node/node.go

```diff
@@ -1330,7 +1330,7 @@ func (node *AlgorandFullNode) VotingKeys(votingRound, keysRound basics.Round) []
 		matchingAccountsKeys[part.Address()] = true
 
 		// Make sure the key is registered.
-		err := node.accountManager.Registry().Register(part.ID(), keysRound)
+		err := node.accountManager.Registry().Register(part.ID(), votingRound)
```

Comment:
Also found a bug, this was causing the wrong "EffectiveRound" variables to be set in the registry.